app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/galleryuplift/GalleryUpliftFragment.kt

```diff
@@ -0,0 +1,720 @@
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.SimpleItemAnimator
+import androidx.viewpager2.widget.ViewPager2
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.google.android.material.chip.Chip
+import dev.arkbuilders.arkfilepicker.folders.RootAndFav
+import dev.arkbuilders.arklib.ResourceId
+import dev.arkbuilders.arklib.data.index.Resource
+import dev.arkbuilders.arklib.data.index.ResourceIndex
+import dev.arkbuilders.arklib.data.meta.Metadata
+import dev.arkbuilders.arklib.user.tags.Tag
+import dev.arkbuilders.arklib.user.tags.TagStorage
+import dev.arkbuilders.arklib.user.tags.Tags
+import dev.arkbuilders.arklib.utils.extension
+import dev.arkbuilders.components.databinding.ScoreWidgetBinding
+import dev.arkbuilders.components.scorewidget.ScoreWidget
+import dev.arkbuilders.navigator.BuildConfig
+import dev.arkbuilders.navigator.R
+import dev.arkbuilders.navigator.data.stats.StatsStorage
+import dev.arkbuilders.navigator.data.utils.LogTags
+import dev.arkbuilders.navigator.databinding.FragmentGalleryBinding
+import dev.arkbuilders.navigator.databinding.PopupGalleryTagMenuBinding
+import dev.arkbuilders.navigator.presentation.App
+import dev.arkbuilders.navigator.presentation.dialog.DetailsAlertDialog
+import dev.arkbuilders.navigator.presentation.dialog.StorageExceptionDialogFragment
+import dev.arkbuilders.navigator.presentation.dialog.edittags.EditTagsDialogFragment
+import dev.arkbuilders.navigator.presentation.navigation.Screens
+import dev.arkbuilders.navigator.presentation.screen.gallery.GalleryFragment
+import dev.arkbuilders.navigator.presentation.screen.gallery.galleryuplift.state.GallerySideEffect
+import dev.arkbuilders.navigator.presentation.screen.gallery.galleryuplift.state.GalleryState
+import dev.arkbuilders.navigator.presentation.screen.main.MainActivity
+import dev.arkbuilders.navigator.presentation.utils.FullscreenHelper
+import dev.arkbuilders.navigator.presentation.utils.extra.ExtraLoader
+import dev.arkbuilders.navigator.presentation.utils.makeGone
+import dev.arkbuilders.navigator.presentation.utils.makeVisible
+import dev.arkbuilders.navigator.presentation.view.DefaultPopup
+import dev.arkbuilders.navigator.presentation.view.DepthPageTransformer
+import dev.arkbuilders.navigator.presentation.view.StackedToasts
+import kotlinx.coroutines.launch
+import org.orbitmvi.orbit.viewmodel.observe
+import timber.log.Timber
+import java.nio.file.Path
+import javax.inject.Inject
+import kotlin.system.measureTimeMillis
+
+class GalleryUpliftFragment : Fragment() {
+    private val binding by viewBinding(FragmentGalleryBinding::bind)
+
+    @Inject
+    lateinit var factory: GalleryUpliftViewModelFactory.Factory
+    private val viewModel: GalleryUpliftViewModel by viewModels {
+        factory.create(
+            requireArguments().getBoolean(SELECTING_ENABLED_KEY)
+        )
+    }
```

Comment:
Look at the [previous version](https://github.com/ARK-Builders/ARK-Navigator/blob/fc2ffe283cc9966fbe0f1098b06652fb98e82c7d/app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/GalleryFragment.kt#L64), there are clearly not enough arguments here

Reply:
Actually I extracted these parameters out of the constructors and put in `initlialize` method. But it looks like that's not the proper way, let me move them back to constructor